### PR TITLE
Code structure and fixes for contact and newsletter form

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: AngularApplication/src/app/core/core.module.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { NewsletterSubscriptionService } from './newsletter-subscription.service';
+
+@NgModule({
+  imports: [
+    CommonModule
+  ],
+  declarations: [],
+  providers: [ NewsletterSubscriptionService ]
+})
+export class CoreModule { }
```
Comment (optimusway) [inline AngularApplication/src/app/core/core.module.ts:12]: why do you need an empty class here? what are responsibilities of this class?

File: AngularApplication/src/app/core/newsletter-subscription.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable()
+export class NewsletterSubscriptionService {
+
+  constructor(private http: HttpClient) { }
```
Comment (optimusway) [inline AngularApplication/src/app/core/newsletter-subscription.service.ts:9]: i believe `private readonly http` would be better, but it depends on the TS version you use

File: AngularApplication/src/app/core/newsletter-subscription.service.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  makeSubscription(newsletterInfo: object): Observable<any> {
```
Comment (optimusway) [inline AngularApplication/src/app/core/newsletter-subscription.service.ts]: is it enough to have just a generic `object` type for the `newsletter`? Does it make sense to have a strongly type param here? The same for the `Observable<any>`

File: AngularApplication/src/app/core/newsletter-subscription.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    return this.http.post(
+      '/api/newsLetter',
+      Object.assign({}, newsletterInfo, {source: 'PartnerPortal'}),
```
Comment (optimusway) [inline AngularApplication/src/app/core/newsletter-subscription.service.ts]: i got 2 points for the `Object.assign`
1) you may want to use TS rest/spread operator here, like `{...newsletterinfo, source: 'source'}`
2) do you really need `Object.assign`/spread here? it has some side effects

File: AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts [hunk 5]
```diff
@@ -71,12 +71,16 @@ export class ContactUsPopupComponent {
...
-      .subscribe(val => {
+    this.homeService.sendContactUs(this.contactUsForm.value)
+    .subscribe(
+      val => {
         this.ready = true;
         this.showSuccessMessage = true;
-      });
+      },
+      error => {
+        this.ready = true;
```
Comment (optimusway) [inline AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts]: not a big expert in rxjs but can we use `finally` section and put `this.ready =true` there?

File: AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts [hunk 6]
```diff
@@ -71,12 +71,16 @@ export class ContactUsPopupComponent {
...
+        alert('An error has occurred. We are sorry for the inconvenience.');
```
Comment (optimusway) [inline AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts]: alert? really?

File: AngularApplication/src/app/home/home.service.ts [hunk 7]
```diff
@@ -11,4 +11,13 @@ export class HomeChartService {
       `assets/data/${assetId}USD.json`
     );
   }
+
+  sendContactUs(contactFormData) {
+    return this.http.post(
+      '/api/contacts/sendContact',
+      Object.assign({}, contactFormData, {source: 'PartnerPortal'}),
```
Comment (optimusway) [inline AngularApplication/src/app/home/home.service.ts]: same as above about using Object.assign/spread

File: AngularApplication/src/app/shared/components/footer/footer.component.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'lpp-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.scss']
+})
+export class FooterComponent implements OnInit {
+
+  constructor() { }
```
Comment (optimusway) [inline AngularApplication/src/app/shared/components/footer/footer.component.ts]: why an empty constructor and an empty`ngOnInit` here?

**Review by optimusway — COMMENTED** [review]
Plz find my comments below

One more general comment is that i saw some logic changes, like migrating from the direct usage of http client to the service inject but couldn't find tests describing that

**Review by KonstantinRyazantsev — APPROVED** [review]
Approved, but comments made by @optimusway should be fixed

File: AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts (hunk 6 above)
Comment (ghost) [reply AngularApplication/src/app/home/contact-us/contact-us-popup/contact-us-popup.component.ts]: This is a temporary solution. Later on in the next pull request we introduce a messaging service for simple alerts like this.

File: AngularApplication/src/app/shared/components/footer/footer.component.ts (hunk 8 above)
Comment (ghost) [reply AngularApplication/src/app/shared/components/footer/footer.component.ts]: It's just a side effect of the Angular CLI generating a component. I will clean this up for the sake of the clean code.

File: AngularApplication/src/app/core/core.module.ts (hunk 1 above)
Comment (ghost) [reply AngularApplication/src/app/core/core.module.ts:12]: This `CoreModule` will be used to hold and provide any services that are needed throughout the application and do not belong to any particular other module. This approached is based on the following Angular documentation article: https://angular.io/guide/ngmodule#the-core-module Please comment if you would like us to amend it in any way.

File: AngularApplication/src/app/core/newsletter-subscription.service.ts (hunk 4 above)
Comment (ghost) [reply AngularApplication/src/app/core/newsletter-subscription.service.ts]: I agree with the suggestion to use the spread operator and will switch to it. Do you have in mind any other approach in this particular situation?

File: AngularApplication/src/app/core/newsletter-subscription.service.ts (hunk 2 above)
Comment (ghost) [reply AngularApplication/src/app/core/newsletter-subscription.service.ts:9]: The app is currently compiling with Typescript v2.4.2. Generally having `readonly` modifier or not, will produce exact same ES5 code once compiled. Having it here with the Angular DI, I don't consider it as necessary. Please confirm if you have established code practice to inject services with `readonly` modifier.

Comment (optimusway) [reply AngularApplication/src/app/core/newsletter-subscription.service.ts:9]: it's all about developer experience and better utilization of tsc, not about transpiled code. It's not critical as well.

DI practice comes from the OOP so you may find best practices here and they use readonly props in classes.

I would suggest to use `readonly` modifier for _injected_ whatever but it's up to you.

**Review by optimusway — APPROVED** [review]
Now it's better!

**Pull request merged**